### PR TITLE
chapter_7

### PR DESCRIPTION
**Pull request by muizzuddinyafiq**
ready

### CODE REVIEW

**Comment by muizzuddinyafiq** [comment]
ready

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -1,2 +1,33 @@
-<h1>Users#new</h1>
-<p>Find me in app/views/users/new.html.erb</p>
+<% provide :title, t("sign_up") %>
+<h1>Sign up</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @user, url: signup_path do |f| %>
+
+    <% if @user.errors.any? %>
+<div id="error_explanation">
```
Comment (ngocvt-0484) [inline app/views/users/new.html.erb]: wrong indent

File: config/routes.rb [hunk 2]
```diff
@@ -1,5 +1,13 @@
 Rails.application.routes.draw do
-  resourcers :users
+  get 'users/show'
+  get 'users/new'
```
Comment (ngocvt-0484) [inline config/routes.rb]: remove it

File: config/routes.rb [hunk 3]
```diff
@@ -1,5 +1,13 @@
 Rails.application.routes.draw do
-  resourcers :users
+  get 'users/show'
```
Comment (ngocvt-0484) [inline config/routes.rb]: remove it

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,13 @@
...
+  resources :users
   root 'static_pages#home'
   get 'static_pages/help'
+
+  get "/signup", to: "users#new"
+  post "/signup", to: "users#create"
+  resources :users, only: %i(new create)
 end
+
+
```
Comment (ngocvt-0484) [inline config/routes.rb]: delete empty line

**Comment by muizzuddinyafiq** [comment]
ready

**Comment by ngocvt-0484** [comment]
why you delete app/assets/stylesheets/application.css 
please keep it

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,4 +1,26 @@
 class UsersController < ApplicationController
+  def show
+    @user = User.find_by id: params[:id]
+  end
+
   def new
+     @user = User.new
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb]: wrong indent

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -1,2 +1,35 @@
...
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
+      <li><%= msg %></li>
+    <% end %>
+    </ul>
+  </div>
+<% end %>
+
+
```
Comment (ngocvt-0484) [inline app/views/users/new.html.erb]: remove blank line

File: config/routes.rb (hunk 3 above)
Comment (ngocvt-0484) [reply config/routes.rb]: not fix yet

File: config/routes.rb (hunk 2 above)
Comment (ngocvt-0484) [reply config/routes.rb]: not fix yet

**Comment by muizzuddinyafiq** [comment]
> why you delete app/assets/stylesheets/application.css
> please keep it

i already rename the application.css to scss, so if i keep the application.css and scss (2 file) , my rails server cannot be open.

**Comment by muizzuddinyafiq** [comment]
ready

File: app/views/users/new.html.erb [hunk 7]
```diff
@@ -1,2 +1,32 @@
...
+      <%= f.label :password %>
+      <%= f.password_field :password %>
+      <%= f.label :password_confirmation, t("confirmation") %>
+      <%= f.password_field :password_confirmation %>
+      <%= f.submit ("Create my account"), class: "btn btn-primary" %>
+    <% end %>
+  </div>
+</div>
+
+<% if @user.errors.any? %>
```
Comment (huongnt-2545) [inline app/views/users/new.html.erb]: create new file named `_error_messages.html.erb` in  `app/views/shared/` folder
move line 19 -> 32 into the file just is created

File: config/routes.rb [hunk 8]
```diff
@@ -1,5 +1,11 @@
 Rails.application.routes.draw do
-  resourcers :users
+  get 'users/show'
```
Comment (huongnt-2545) [inline config/routes.rb]: remove this line

File: config/routes.rb [hunk 9]
```diff
@@ -1,5 +1,11 @@
...
+  get 'users/new'
```
Comment (huongnt-2545) [inline config/routes.rb]: remove this line

File: config/routes.rb [hunk 10]
```diff
@@ -1,5 +1,11 @@
...
+
   root 'static_pages#home'
   get 'static_pages/help'
+
+  get "/signup", to: "users#new"
+  post "/signup", to: "users#create"
+  resources :users, only: %i(new create)
```
Comment (huongnt-2545) [inline config/routes.rb]:
```suggestion
  resources :users, only: %i(new create show)
```

**Comment by muizzuddinyafiq** [comment]
ready

File: config/routes.rb (hunk 3 above)
Comment (ngocvt-0484) [reply config/routes.rb]: not fix yet

**Comment by ngocvt-0484** [comment]
please fix my comments before ready

**Comment by ngocvt-0484** [comment]
image

**Comment by muizzuddinyafiq** [comment]
ready

File: app/assets/stylesheets/application.css [hunk 11]
```diff
@@ -1,15 +0,0 @@
...
- * listed below.
- *
- * Any CSS and SCSS file within this directory, lib/assets/stylesheets, or any plugin's
- * vendor/assets/stylesheets directory can be referenced here using a relative path.
- *
- * You're free to add application-wide styles to this file and they'll appear at the bottom of the
- * compiled file so the styles you add here take precedence over styles defined in any other CSS/SCSS
- * files in this directory. Styles in this file should be added after the last require_* statement.
- * It is generally better to create a new file per style scope.
- *
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/application.css:12]: keep application.scss 
not delete it

**Comment by ngocvt-0484** [comment]
not fix comment yet

**Pull request merged**